modules/sdk/src/main/scala/org/tessellation/sdk/domain/consensus/ConsensusFunctions.scala

```diff
@@ -17,6 +17,12 @@ trait ConsensusFunctions[F[_], Event, Key, Artifact, Context] {
     artifact: Artifact
   ): F[Either[InvalidArtifact, Artifact]]
 
+  def createContext(
+    lastSnapshotContext: Context,
+    lastSnapshot: Artifact,
+    snapshot: Signed[Artifact]
```

Comment:
If this is in `ConsensusFunctions`, I'd keep the original naming like `key`, `context` and `artifact`. it doesn't need to be a snapshot